### ISSUE TITLE
Invalid date format in response header causes the whole request to fail

### ISSUE THREAD

**Issue by amatriain** [rtomayko/rack-cache#107]
I'm using RestClient along with rest-client-components and rack-cache to perform client-side caching.

I've found that when I make a GET request to the same URL for the second and consecutive times, if the "Expires" header in the response has an invalid date format, an ArgumentError is raised and the whole request fails. Is there a way to make rack-cache ignore the invalid header as if it wasn't present, instead of raising an error?

This is an example of an URL that causes this: http://www.asoc-acat.es/portal/index.php?format=feed&type=rss

The `Expires` header in the response has this value:

```
Expires Mon, 1 Jan 2001 00:00:00 GMT
```

which is an invalid date because according to RFC2616 the day should have two digits, it should be like this: 

```
Expires Mon, 01 Jan 2001 00:00:00 GMT
```

When I try to GET from this URL with rack-cache enabled, the following is raised:

```
ArgumentError: not RFC 2616 compliant date: "Mon, 1 Jan 2001 00:00:00 GMT"
```

with this backtrace:

```
 "~/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/time.rb:544:in `httpdate'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/response.rb:165:in `expires'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/response.rb:160:in `max_age'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/response.rb:185:in `ttl'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/response.rb:93:in `fresh?'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:125:in `fresh_enough?'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:175:in `lookup'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:66:in `call!'"
 "~/.rvm/gems/ruby-2.2.0/gems/rack-cache-1.2/lib/rack/cache/context.rb:51:in `call'"
 "~/.rvm/gems/ruby-2.2.0/gems/rest-client-components-1.3.0/lib/restclient/components.rb:12:in `call'"
 "~/.rvm/gems/ruby-2.2.0/gems/rest-client-components-1.3.0/lib/restclient/components.rb:136:in `execute'"
 "~/.rvm/gems/ruby-2.2.0/gems/rest-client-1.7.2/lib/restclient/request.rb:41:in `execute'"
 "~/.rvm/gems/ruby-2.2.0/gems/rest-client-1.7.2/lib/restclient.rb:65:in `get'"
```

If I'm reading it correctly, the error is raised when rack-cache tries to use the `Time.httpdate` method to convert the `Expires` header into a date object to decide if the cache is still fresh. 

In my opinion raising an error that makes the whole GET invocation fail is too harsh for such a small error. I think it would be better if rack-cache just ignored the invalid header and processed the request as if no `Expires` header was present in the response. Is there a way to achieve this with rack-cache?

**Comment by boone** [comment]
This happened to me too. I submitted a PR in #97. Since this project appears dormant, I am just using a fork for now.

**Comment by amatriain** [comment]
@boone thanks for pointing me to your PR. I've also forked the repo and applied your commit, it's very straightforward and I see no problem with it.

I'm leaving this issue open in case the repo owner comes back.

**Comment by amatriain** [comment]
This issue was fixed by @boone in PR #97 only for the case in which the Expires header is unparseable. If the Date header is unparseable the same error will happen.

I've submitted PR #146 that applies a similar fix to the Date header.
